usr/sbin/pkcsconf/pkcsconf.c

```diff
@@ -105,47 +100,13 @@ int main(int argc, char *argv[])
             }
             break;
         case 'S':              /* the SO pin */
-            if (flags & CFG_SO_PIN) {
-                warnx("Must specify a single SO PIN.");
-                errflag++;
-            } else {
-                flags |= CFG_SO_PIN;
-                sopin = (CK_CHAR_PTR) malloc(strlen(optarg) + 1);
-                if (sopin == NULL) {
-                    rv = CKR_HOST_MEMORY;
-                    goto done;
-                }
-                memcpy(sopin, optarg, strlen(optarg) + 1);
-            }
+            pin_so = optarg;
```

Comment:
OK its nit picking, but you loose the ability to detect that the -S flag is specified multiple times.  Maybe you should check if pin_so != NULL and issue the error message if so.

Same for -U and -n below.

Reply:
ok, I'll fix this.